python/cugraph/cugraph/gnn/data_loading/bulk_sampler_io.py

```diff
@@ -63,24 +67,73 @@ def _write_samples_to_parquet(
         if end_batch_id == max_batch_id:
             end_ix = len(results)
         else:
-            end_ix = offsets.offsets[offsets.batch_id == (end_batch_id + 1)].iloc[0]
+            offsets_z = offsets[offsets.batch_id == (end_batch_id + 1)]
+            end_ix = offsets_z.offsets.iloc[0]
 
         full_output_path = os.path.join(
             output_path, f"batch={start_batch_id}-{end_batch_id}.parquet"
         )
-        results_p = results.iloc[start_ix:end_ix]
+        results_p = results.iloc[start_ix:end_ix].reset_index(drop=True)
 
         results_p["batch_id"] = offsets_p.batch_id.repeat(
             cupy.diff(offsets_p.offsets.values, append=end_ix)
         ).values
-        results_p.to_parquet(full_output_path, compression=None, index=False)
+
+        if renumber_map is not None:
+            renumber_map_start_ix = offsets_p.renumber_map_offsets.iloc[0]
+
+            if end_batch_id == max_batch_id:
+                renumber_map_end_ix = len(renumber_map)
+            else:
+                renumber_map_end_ix = offsets_z.renumber_map_offsets.iloc[0]
```

Comment:
QQ: I dont understand this logic. Why is `renumber_map_end_ix ` being set to the same value as `  renumber_map_start_ix` ?

Reply:
Sorry for the logic, it's not the same value.  `offsets_p` contains the start offsets and `offset_z` contains the end offsets.

Reply:
Ahh, Cool. That makes sense.

Reply:
Lets maybe rename those variables to reflect `start` and `end` instead of `p` and `z`